GeneratingDataset.py

```diff
@@ -1051,16 +1078,65 @@ def get_audio_features(self, audio, sample_rate):
       assert feature_data.shape[1] == self.get_feature_dimension()
 
     if self.norm_mean is not None:
-      feature_data -= self.norm_mean[None, :]
+      if isinstance(self.norm_mean, (int, float)):
+        feature_data -= self.norm_mean
+      else:
+        feature_data -= self.norm_mean[None, :]
+
     if self.norm_std_dev is not None:
-      feature_data /= self.norm_std_dev[None, :]
+      if isinstance(self.norm_std_dev, (int, float)):
+        feature_data /= self.norm_std_dev
+      else:
+        feature_data /= self.norm_std_dev[None, :]
+
+    if self.join_frames is not None:
+      pad_len = self.join_frames - (feature_data.shape[0] % self.join_frames)
+      pad_len = pad_len % self.join_frames
+      new_len = feature_data.shape[0] + pad_len
+      feature_data = numpy.pad(feature_data, pad_width=((0, pad_len), (0, 0)), mode="edge")
+      feature_data = numpy.reshape(feature_data,
+                                   newshape=(int(new_len/self.join_frames), feature_data.shape[1]*self.join_frames),
+                                   order='C')
+
     return feature_data
 
   def get_feature_dimension(self):
     """
     :rtype: int
     """
-    return (self.with_delta + 1) * self.num_feature_filters
+    if self.join_frames is not None:
+      return ((self.with_delta + 1) * self.num_feature_filters) * self.join_frames
+    return ((self.with_delta + 1) * self.num_feature_filters)
```

Comment:
I think it's cleaner to not duplicate code, but to write this instead:
`return return ((self.with_delta + 1) * self.num_feature_filters) * (self.join_frames or 1)`

Reply:
(Also, remove the unnecessary brackets.)